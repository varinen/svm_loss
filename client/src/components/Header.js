import React from 'react';

const Header = () => (
    <header className="masthead bg-white">
        <div className="mx-auto text-center">
            <h1 className="mx-auto my-4 text-uppercase text-gray-700">SVM
                Loss</h1>
            <h2 className="text-gray-900 mx-auto mt-2 mb-5">
                Linear Classifier Demo
            </h2>

        </div>
    </header>
);

export default Header;