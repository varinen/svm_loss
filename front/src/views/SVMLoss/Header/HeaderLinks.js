/*eslint-disable*/
import React from "react";
import DeleteIcon from "@material-ui/icons/Delete";
import IconButton from "@material-ui/core/IconButton";
// react components for routing our app without refresh
import { Link } from "react-router-dom";

// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import Tooltip from "@material-ui/core/Tooltip";

// @material-ui/icons
import { Apps, CloudDownload } from "@material-ui/icons";

// core components
import CustomDropdown from "components/CustomDropdown/CustomDropdown.js";
import Button from "components/CustomButtons/Button.js";

import styles from "assets/jss/material-kit-react/components/headerLinksStyle.js";

const useStyles = makeStyles(styles);

export default function HeaderLinks(props) {
  const classes = useStyles();
  return (
    <List className={classes.list}>

      <ListItem className={classes.listItem}>
        <Tooltip
          id="github"
          title="Github"
          placement={window.innerWidth > 959 ? "top" : "left"}
          classes={{ tooltip: classes.tooltip }}
        >
          <Button
            color="transparent"
            href="https://github.com/varinen/svm_loss"
            target="_blank"
            className={classes.navLink}
          >
            <i className={classes.socialIcons + " fab fa-github"} />
          </Button>
        </Tooltip>
      </ListItem>

    </List>
  );
}
