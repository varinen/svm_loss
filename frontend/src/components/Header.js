import React, {Fragment} from "react";
import {Typography} from "@material-ui/core";
import useStyles from "../styles";

export default () => {
    const classes = useStyles();
    return (
        <Fragment>
            <Typography variant="h2" component="h1" gutterBottom className={classes.heading}>
                SVM Loss
            </Typography>
            <Typography variant="h5" component="h2" gutterBottom className={classes.heading}>
                Linear Classifier Demo
            </Typography>
        </Fragment>
    )
}
