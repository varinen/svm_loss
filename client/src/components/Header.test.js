import React from 'react';
import {shallow} from 'enzyme';
import Header from './Header';

describe('Header', () => {
    const header = shallow(<Header/>);

    it('renders correctly', () => {
        expect(header).toMatchSnapshot();
    });
});

