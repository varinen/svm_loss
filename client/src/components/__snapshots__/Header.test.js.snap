// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="masthead bg-white"
>
  <div
    className="mx-auto text-center"
  >
    <h1
      className="mx-auto my-4 text-uppercase text-gray-700"
    >
      SVM Loss
    </h1>
    <h2
      className="text-gray-900 mx-auto mt-2 mb-5"
    >
      Linear Classifier Demo
    </h2>
  </div>
</header>
`;
